pkg/signup/service/signup_service.go

```diff
@@ -461,6 +465,30 @@ func (s *ServiceImpl) DoGetSignup(ctx *gin.Context, provider ResourceProvider, u
 		Message:              murCondition.Message,
 		VerificationRequired: states.VerificationRequired(userSignup),
 	}
+
+	if mur.Status.ProvisionedTime != nil {
+		startDateValue := mur.Status.ProvisionedTime.Format(ISO8601Format)
+		signupResponse.StartDate = &startDateValue
+
+		// #### The rest of this code only makes sense when there is a ProvisionedTime set, so we include it in the same block ####
+
+		// Lookup the user's tier
+		tier, err := provider.GetUserTier(mur.Spec.TierName)
```

Comment:
Let's use the informer and informer service to get the tier. We also can switch to the informer service to obtain the mur: https://github.com/codeready-toolchain/registration-service/blob/master/pkg/informers/service/informer_service.go

It should work much faster. Thanks to the informer cache.

And don't forget to update the reg-service SA's role so it has permissions to list/get/watch tiers.
For dev environment: https://github.com/codeready-toolchain/registration-service/blob/master/deploy/registration-service.yaml#L33
For prod & e2e-tests: https://github.com/codeready-toolchain/host-operator/blob/master/deploy/registration-service/registration-service.yaml#L33

Reply:
Thanks for the quick initial review, I've implemented the UserTier informer support, and made all the changes you suggested.  Host operator PR with the new permissions is here: https://github.com/codeready-toolchain/host-operator/pull/988